tests/test_api.py

```diff
@@ -41,6 +42,9 @@ def test_for_name_does_not_exist(self):
         with self.assertRaises(PackageNotFoundError):
             distribution('does-not-exist')
 
+    def test_for_name_containing_dot(self):
+        assert distribution('pkg-dot').metadata['Name'] == 'pkg.dot'
```

Comment:
I think the most important use-case to cover is that `distribution('pkg.dot')` should resolve when the name on the file system is normalized (`pkg_dot`). I'm not sure I want a unit test or declared support for accepting pre-normalized names.

Reply:
In subsequent commits, I agree it's worthwhile to accept normalized names, though I de-emphasize that acceptance in the changelog.